python_modules/dagster/dagster/_config/structured_config/__init__.py

```diff
@@ -1240,3 +1241,61 @@ def validate_resource_annotated_function(fn) -> None:
                 else "'ResourceParam[Any]' or 'ResourceParam[<output type>]'",
             )
         )
+
+
+class IAttachBareObjectToContext:
```

Comment:
IAttachDifferentObjectOnContext?

"Bare" isn't really a meaningful adjective here IMO

Reply:
We could also consider not having the interface at all and instead just having the method `get_object_to_set_on_execution_context` to avoid exporting another public symbol.

However I do like the marker interface and it makes it clear that this is a temporary, compartmentalized measure. (We should move it to its own file)